### PR TITLE
[CHILI GraFx] Custom Media Connectors

### PR DESCRIPTION
**Pull request by BramVerniest**

### CODE REVIEW

**Review by brapoprod — APPROVED** [review]

File: docs/GraFx-Developers/connectors/build-media-connector/index.md [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+```shell
++35.619s CLI authenticated successfully.
++35.812s User is authenticated => demo@test.com
+
+```
+
+When that is done, you can now publish.
+
+``` zsh
+npx connector-cli publish -e <environment> -b <base url> -n <connector name>
```
Comment (psamusev) [inline docs/GraFx-Developers/connectors/build-media-connector/index.md:264]: @BramVerniest  in context of this PR https://github.com/chili-publish/studio-connector-contrib/pull/15, we've added possibility to specify `runtime-options` as well as proxy-options for the deployed connector. Moreover it allows you have `required` and `optional` `runtime-options` . As a result you won't be able to publish without `required` options when doing `publish` command. Might be mentioned some of them here, I think.

File: docs/GraFx-Developers/connectors/build-media-connector/index.md [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+``` csh
+npx connector-cli
+```
+
+
+Initialize: Means building the Boiler Plate. This is the basic code you'll need to get started.
+
+``` csh
+npx connector-cli init
```
Comment (psamusev) [inline docs/GraFx-Developers/connectors/build-media-connector/index.md:48]: In context of https://github.com/chili-publish/studio-connector-contrib/pull/16, we've added possibility to specify `generated directory` to be able use not only the `current one`. Like `npx connector-cli init ./connectors/connector-1`. If it's worth to mention, please specify it somewhere here

File: docs/GraFx-Developers/connectors/build-media-connector/index.md [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+
+found 0 vulnerabilities
+
+```
+
+Now it's time to create your custom code.
+
+This happens in the file "connector.ts". This is a typescript files.
+
+By default, there will be no implementations. Your IDE can do this for you, to build the default interface methods.
```
Comment (psamusev) [inline docs/GraFx-Developers/connectors/build-media-connector/index.md:74]: In context of https://github.com/chili-publish/studio-connector-contrib/pull/16, we've added `default` implementation, so you might skip the next step

File: docs/GraFx-Developers/connectors/build-media-connector/index.md [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+
+```zsh
+npx connector-cli build
+```
+
+resulting in
+
+```zsh
++0.045s connector-cli v1.0.49
++0.046s Running command: 'build' with options: {"connectorFile":"./connector.ts","options":{}}
```
Comment (psamusev) [inline docs/GraFx-Developers/connectors/build-media-connector/index.md:132]: in context of https://github.com/chili-publish/studio-connector-contrib/pull/15, we hide `connector-cli v` and `Running command...` output (as it's debug info). To be able see them, you need explicitly path `-v` (verbose) flag. I.e. `npc connector-cli build -v`. But again, please consider `-v` as something when you need to debug the behaviour. It applies for other commands as well

File: docs/GraFx-Developers/connectors/build-media-connector/index.md (hunk 1 above)
Comment (BramVerniest) [reply docs/GraFx-Developers/connectors/build-media-connector/index.md:264]: @psamusev Will write down as future improvement. But would like to get v1 of this docs out

File: docs/GraFx-Developers/connectors/build-media-connector/index.md (hunk 2 above)
Comment (BramVerniest) [reply docs/GraFx-Developers/connectors/build-media-connector/index.md:48]: Will be for future update

File: docs/GraFx-Developers/connectors/build-media-connector/index.md (hunk 4 above)
Comment (BramVerniest) [reply docs/GraFx-Developers/connectors/build-media-connector/index.md:132]: Future update

**Review by psamusev — APPROVED** [review]

**Pull request merged**